modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/GlassViewEventHandler.java

```diff
@@ -675,6 +675,12 @@ public double[] getInputMethodCandidatePos(int offset) {
         return ret;
     }
 
+    @Override
+    public double[] getInputMethodCandidateRelativePos(int offset) {
+        Point2D p2d = scene.inputMethodRequests.getTextLocationRelative(offset);
+        return new double[] { p2d.getX(), p2d.getY() };
```

Comment:
On my system the IM window is incorrectly positioned. This appears to be because I'm running a high-DPI display with 200% magnification. I think you need to call getPlatformScaleX and getPlatformScaleY and apply those scaling factors before passing these coordinates on to glass. You'll see other places in this file where conversions like that are being performed.

Reply:
I did revert the relative positioning change as it will get in the way of merging this. Could you check if it's correctly positioned now? I don't own a fancy monitor :) Tried 200% scale here and everything looks monstrous. 125% scale looks correct.